sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -2316,7 +2328,9 @@ private void SetNameFieldsIfNull()
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+#pragma warning disable AZC0002 // Client method should have an optional CancellationToken (both name and it being optional matters) or a RequestContext as the last parameter.
```

Comment:
I am happy to see these being added in line, rather than in a GlobalSuppressions file, because it means the analyzer will still catch issues in the future.  Out of curiosity, what has changed in the analyzer for AZC0002 that made it start catching these where it didn't before?  For these overloads that call through to a method that takes a cancellation token, could we identify those in the analyzer?  I think the goal of the analyzer is to ensure that there is an overload that takes a cancellation token in the public API.

Reply:
Yeah, I changed the logic to look for overloads. Then the suppression could be removed.